### PR TITLE
[4.3] Missing php fileinfo

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #39711 .

### Summary of Changes
When the php fileinfo module is not installed then media manager will give the following error

> Error Property 'mime_type' of argument 'item' of event onFetchMediaItems has a wrong item. Valid: string

This PR adds a check for the php fileinfo module and displays a more helpful error message

> The Fileinfo PHP module is not installed. You should contact the server administrator to install it.

### Testing Instructions
To display the error message
Either disable the fileinfo module in your php.ini and open the media manager
or edit the check to reverse it so it shows when the module is present by removing the ! in (!\extension_loaded('fileinfo')) {

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

There may be changes required in the docblock - no idea

### CODE REVIEW

File: administrator/components/com_media/src/Event/AbstractMediaItemValidationEvent.php [hunk 1]
```diff
@@ -52,6 +53,11 @@ abstract class AbstractMediaItemValidationEvent extends AbstractImmutableEvent
      */
     protected function validate(\stdClass $item): void
     {
+        // Before any of the following checks we should first check that the php module fileinfo is installed
+        if (!\extension_loaded('fileinfo')) {
```
Comment (sandewt) [inline administrator/components/com_media/src/Event/AbstractMediaItemValidationEvent.php:57]: ~~`fileinfo` ?~~
~~or~~
~~`phpinfo` ?~~

Comment (brianteeman) [reply administrator/components/com_media/src/Event/AbstractMediaItemValidationEvent.php:57]: fileinfo is correct. we are testing to see if the extension is loaded

**Comment by sandewt** [comment]
I have tested this item :white_check_mark: successfully on 069e062f5977b25daec5240f6a1ec9720a16c5c8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39777">issues.joomla.org/tracker/joomla-cms/39777</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 069e062f5977b25daec5240f6a1ec9720a16c5c8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39777">issues.joomla.org/tracker/joomla-cms/39777</a>.</sub>

File: administrator/components/com_media/src/Event/AbstractMediaItemValidationEvent.php (hunk 1 above)
Comment (sandewt) [reply administrator/components/com_media/src/Event/AbstractMediaItemValidationEvent.php:57]: :bulb:

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39777">issues.joomla.org/tracker/joomla-cms/39777</a>.</sub>

**Comment by SharkyKZ** [comment]
This makes no sense.

**Comment by brianteeman** [comment]
> This makes no sense.

maybe not but would you care to explain why, please

**Comment by bembelimen** [comment]
There are several "problems" with this check.
1. This added check does not validate the file (what the event is about) but it validates the system it's running on.
2. You assume, that the file is generated via the LocalAdapter, which is not always the case
3. There are [more mime checker](https://github.com/joomla/joomla-cms/blob/4.3-dev/libraries/src/Helper/MediaHelper.php#L86) than "fileinfo", so although this could be disabled, others could work
4. The error is not in this plugin but you have to fix it [here](https://github.com/joomla/joomla-cms/blob/4.3-dev/plugins/filesystem/local/src/Adapter/LocalAdapter.php#L392)

**Comment by brianteeman** [comment]
Such an unfriendly reply. Not worth my time explaining where you are mistaken

**Comment by HLeithner** [comment]
> Such an unfriendly reply. Not worth my time explaining where you are mistaken

Tbh but this was unfriendly and not that Benjamin explained you why this pr is wrong in a normal language.

**Comment by brianteeman** [comment]
wouldnt expect anything less

**Comment by C-Lodder** [comment]
Incase anyone decides to work on this, might also be worth adding it to the recommended settings on the pre-update check page:

image

**Comment by brianteeman** [comment]
According to @bembelimen there are better ways to do this and do not need the fileinfo module

**Comment by jfeedback** [comment]
Making sure I've got this right.  The code now relies on a PHP extension which isn't [documented as required](https://downloads.joomla.org/technical-requirements) with a message proposed to at least inform users about things not working if that extension isn't installed 76 days before release.  It was shot down as "not right", with a senior developer saying there are better solutions, and in those 76 days, nobody could come up with any of those better solutions.  And, everyone is OK with this?

**Comment by obuisard** [comment]
@jfeedback  Granted that we need to find a suitable answer to this problem as soon as possible. However, shaming the team of developers is surely not going to help. You are welcome to contribute to a solution, even if you are not a developer (as a user, what would you consider a helpful solution here?).

**Pull request closed without merging**